### PR TITLE
[Homepage] Fix for analytics bug

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
We have a GA event on the homepage that is firing sometimes with empty values. For some reason, the event listener wasn't tied directly to the anchor tag, so clicking space around the tag would trigger the event. This PR fixes that by just attaching the `onclick` event directly to the anchor tag.

https://github.com/department-of-veterans-affairs/va.gov-team/issues/727

## Testing done
- Clicked around the anchor tag on the homepage confirming that the event only triggers on a direct click

## Screenshots
Google Analytics Debugger

image

## Acceptance criteria
- [ ] Correct GTM event is captured only when event occurs
 
## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/layouts/home.drupal.liquid [hunk 1]
```diff
@@ -14,8 +14,18 @@
...
           <h2 class="heading-level-3 hub-links-title"><i class="icon-large-baseline icon-heading hub-icon-{{ card.label | downcase | replace: ' ', '-' }} hub-color-{{ card.label | downcase | replace: ' ', '-' }}"></i>{{ card.label }}</h2>
           <ul class="hub-links-list">
+            {% assign cardLabel = card.label %}
             {% for link in card.links %}
-              <li><a data-nav-path="{{ card.label }}->{% if link.url.path != empty %}{{ link.url.path }}{% else %}{{ link.label }}{% endif %}" href="{{ link.url.path }}">{{ link.label }}</a></li>
+              <li>
+                <a href="{{ link.url.path }}"
+                  onclick="recordEvent({
+                    event: 'nav-zone-one',
+                    'nav-path': '{{ cardLabel }}->{% if link.url.path != empty %}{{ link.url.path }}{% else %}{{ link.label }}{% endif %}',
```
Comment (kelsonic) [inline src/site/layouts/home.drupal.liquid]: This is subjective, but do you think `{{ card.label }}` to avoid the assignment line above is easier to read?

**Review by kelsonic — APPROVED** [review]
LGTM! 🎉

File: src/site/layouts/home.drupal.liquid (hunk 1 above)
Comment (ncksllvn) [reply src/site/layouts/home.drupal.liquid]: I do, but the Liquid scoping couldn't find `card.label` so it produced the empty string 🤷‍♂

Comment (kelsonic) [reply src/site/layouts/home.drupal.liquid]: Ugggh liquid... 🤦‍♂

**Review by rianfowler — APPROVED** [review]

**Pull request merged**